ambry-cloud/src/main/java/com.github.ambry.cloud/CloudTokenPersistor.java

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright 2019 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.cloud;
+
+import com.github.ambry.clustermap.ClusterMap;
+import com.github.ambry.replication.PartitionInfo;
+import com.github.ambry.replication.ReplicaTokenInfo;
+import com.github.ambry.replication.ReplicaTokenPersistor;
+import com.github.ambry.replication.ReplicationException;
+import com.github.ambry.replication.ReplicationMetrics;
+import com.github.ambry.store.FindTokenFactory;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * {@link CloudTokenPersistor} persists replication token to a cloud storage.
+ */
+public class CloudTokenPersistor extends ReplicaTokenPersistor {
+
+  private static final Logger logger = LoggerFactory.getLogger(CloudTokenPersistor.class);
+  private final String replicaTokenFileName;
+  private final CloudDestination cloudDestination;
+
+  /**
+   * Constructor for {@link CloudTokenPersistor}.
+   * @param replicaTokenFileName the token's file name.
+   * @param partitionGroupedByMountPath A map between mount path and list of partitions under this mount path.
+   * @param replicationMetrics metrics including token persist time.
+   * @param clusterMap the {@link ClusterMap} to deserialize tokens.
+   * @param tokenfactory the {@link FindTokenFactory} to deserialize tokens.
+   */
+  public CloudTokenPersistor(String replicaTokenFileName, Map<String, List<PartitionInfo>> partitionGroupedByMountPath,
+      ReplicationMetrics replicationMetrics, ClusterMap clusterMap, FindTokenFactory tokenfactory, CloudDestination cloudDestination) {
+    super(partitionGroupedByMountPath, replicationMetrics, clusterMap, tokenfactory);
+    this.replicaTokenFileName = replicaTokenFileName;
+    this.cloudDestination = cloudDestination;
+  }
+
+  // Note: assuming that passed mountPath is the partitionId path
+
+  @Override
+  protected void persistTokens(String mountPath, List<ReplicaTokenInfo> tokenInfoList) throws IOException, ReplicationException {
```

Comment:
Could this just be called `persist()` or `write()`? Finding it a little confusing that both `CloudDestination` and `ReplicaTokenPersistor` having the same method names of `persistTokens()` and `retrieveTokens()`

Reply:
I don't have a strong preference so I changed the method names to persist() and retrieve()